api/institutions/authentication.py

```diff
@@ -102,6 +103,10 @@ def authenticate(self, request):
         # unconfirmed, etc.).
         user, created = get_or_create_user(fullname, username, reset_password=False)
 
+        # The `department` field is updated each login/
+        if department:
+            user.department = department
```

Comment:
* `user` is not saved in some conditions. If you take a look at entire flow, you would see that only a new user or a user without an affiliation is saved.
* Should we detect department change and inform both us and the user?